coalib/misc/CachingUtilities.py

```diff
@@ -0,0 +1,174 @@
+import calendar
+import os
+import pickle
+import time
+
+from coalib.output.Tagging import get_user_data_dir
```

Comment:
We should probably move `get_user_data_dir` to misc.Utilities ?
(File a new issue if you agree)